src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -185,14 +186,35 @@ public static Appointment parseAppointment(String appointmentName, String appoin
 
         LocalDateTime appointmentDate;
         try {
-            appointmentDate = Appointment.parseAppointmentDate(appointmentDateString);
+            appointmentDate = parseValidAppointmentDate(appointmentDateString);
         } catch (DateTimeParseException e) {
             throw new ParseException(Appointment.MESSAGE_INVALID_DATE);
         }
 
         return new Appointment(trimmedAppointmentName, appointmentDate);
     }
 
+    /**
+     * Parses a {@code String date} into a {@code LocalDateTime}.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+    public static LocalDateTime parseValidAppointmentDate(String appointmentDateString) throws ParseException {
```

Comment:
Perhaps it would be better to leave it as `parseAppointmentDate`, because this method still throws a `ParseException`. It can't be valid if it still has a risk of throwing an error, right?

Or, maybe `parseAppointmentDateTime` would be better because it returns a LocalDateTime.